classes/migration/SubmissionFilesMigration.inc.php

```diff
@@ -24,63 +23,44 @@ class SubmissionFilesMigration extends Migration {
         public function up() {
 		// Files associated with submission. Includes submission files, etc.
 		Capsule::schema()->create('submission_files', function (Blueprint $table) {
-			$table->bigInteger('file_id')->autoIncrement();
-			$table->bigInteger('revision');
-			$table->bigInteger('source_file_id')->nullable();
-			$table->bigInteger('source_revision')->nullable();
+			$table->bigInteger('submission_file_id')->autoIncrement();
 			$table->bigInteger('submission_id');
-			$table->string('file_type', 255);
+			$table->bigInteger('file_id');
+			$table->bigInteger('source_submission_file_id')->nullable();
 			$table->bigInteger('genre_id')->nullable();
-			$table->bigInteger('file_size');
-			$table->string('original_file_name', 127)->nullable();
 			$table->bigInteger('file_stage');
 			$table->string('direct_sales_price', 255)->nullable();
 			$table->string('sales_type', 255)->nullable();
 			$table->smallInteger('viewable')->nullable();
-			$table->datetime('date_uploaded');
-			$table->datetime('date_modified');
+			$table->datetime('created_at');
+			$table->datetime('updated_at');
 			$table->bigInteger('uploader_user_id')->nullable();
 			$table->bigInteger('assoc_type')->nullable();
```

Comment:
I was about to recommend thinking about alternatives to `assoc_type`/`assoc_id` here, but it turns out that Laravel uses a similar pattern (and `morphs` function). See https://laravel.com/docs/8.x/eloquent-relationships#polymorphic-relationships for details.

Reply:
I think we could get rid of `assoc_type`/`assoc_id` and have some ideas about that for the future. Just one of those things I drew a line under as this refactor got bigger and bigger.